### ISSUE TITLE
Crash when deleting the post on a post list containing only one post

### ISSUE THREAD

**Issue by kahkeshan2** [wordpress-mobile/WordPress-Android#3099]
Whenever I want to delete the last post in my blogposts the application goes down and force me out.This error only happens for the last post. the version which is published in play store doesn't have this problem.

**Comment by maxme** [comment]
> the version which is published in play store doesn't have this problem.

So I presume this only happens on current `develop` branch?

**Comment by kahkeshan2** [comment]
I guess so , please fix it if it's possible.

**Comment by maxme** [comment]
Thanks for the report, I can confirm the crash on `release/4.4` and `develop`:
```
E/AndroidRuntime(  994): FATAL EXCEPTION: main
E/AndroidRuntime(  994): java.lang.IndexOutOfBoundsException: Inconsistency detected. Invalid view holder adapter positionViewHolder{539f4464 position=0 id=-1, oldPos=1, pLpos:1 scrap tmpDetached no parent}
E/AndroidRuntime(  994): 	at android.support.v7.widget.RecyclerView$Recycler.validateViewHolderForOffsetPosition(RecyclerView.java:4349)
E/AndroidRuntime(  994): 	at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:4480)
E/AndroidRuntime(  994): 	at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:4461)
E/AndroidRuntime(  994): 	at android.support.v7.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:1962)
E/AndroidRuntime(  994): 	at android.support.v7.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1371)
E/AndroidRuntime(  994): 	at android.support.v7.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1334)
E/AndroidRuntime(  994): 	at android.support.v7.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:563)
E/AndroidRuntime(  994): 	at android.support.v7.widget.RecyclerView.dispatchLayout(RecyclerView.java:2801)
E/AndroidRuntime(  994): 	at android.support.v7.widget.RecyclerView.onLayout(RecyclerView.java:3145)
E/AndroidRuntime(  994): 	at android.view.View.layout(View.java:13754)
E/AndroidRuntime(  994): 	at android.view.ViewGroup.layout(ViewGroup.java:4362)
E/AndroidRuntime(  994): 	at android.support.v4.widget.SwipeRefreshLayout.onLayout(SwipeRefreshLayout.java:581)
E/AndroidRuntime(  994): 	at android.view.View.layout(View.java:13754)
E/AndroidRuntime(  994): 	at android.view.ViewGroup.layout(ViewGroup.java:4362)
E/AndroidRuntime(  994): 	at android.support.design.widget.CoordinatorLayout.layoutChild(CoordinatorLayout.java:1034)
E/AndroidRuntime(  994): 	at android.support.design.widget.CoordinatorLayout.onLayoutChild(CoordinatorLayout.java:744)
E/AndroidRuntime(  994): 	at android.support.design.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:758)
E/AndroidRuntime(  994): 	at android.view.View.layout(View.java:13754)
E/AndroidRuntime(  994): 	at android.view.ViewGroup.layout(ViewGroup.java:4362)
E/AndroidRuntime(  994): 	at android.widget.RelativeLayout.onLayout(RelativeLayout.java:948)
E/AndroidRuntime(  994): 	at android.view.View.layout(View.java:13754)
E/AndroidRuntime(  994): 	at android.view.ViewGroup.layout(ViewGroup.java:4362)
E/AndroidRuntime(  994): 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1649)
E/AndroidRuntime(  994): 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1507)
E/AndroidRuntime(  994): 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1420)
E/AndroidRuntime(  994): 	at android.view.View.layout(View.java:13754)
E/AndroidRuntime(  994): 	at android.view.ViewGroup.layout(ViewGroup.java:4362)
E/AndroidRuntime(  994): 	at android.widget.FrameLayout.onLayout(FrameLayout.java:448)
E/AndroidRuntime(  994): 	at android.view.View.layout(View.java:13754)
E/AndroidRuntime(  994): 	at android.view.ViewGroup.layout(ViewGroup.java:4362)
E/AndroidRuntime(  994): 	at android.widget.FrameLayout.onLayout(FrameLayout.java:448)
E/AndroidRuntime(  994): 	at android.view.View.layout(View.java:13754)
E/AndroidRuntime(  994): 	at android.view.ViewGroup.layout(ViewGroup.java:4362)
E/AndroidRuntime(  994): 	at android.widget.FrameLayout.onLayout(FrameLayout.java:448)
E/AndroidRuntime(  994): 	at android.view.View.layout(View.java:13754)
E/AndroidRuntime(  994): 	at android.view.ViewGroup.layout(ViewGroup.java:4362)
E/AndroidRuntime(  994): 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1649)
E/AndroidRuntime(  994): 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1507)
E/AndroidRuntime(  994): 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1420)
E/AndroidRuntime(  994): 	at android.view.View.layout(View.java:13754)
E/AndroidRuntime(  994): 	at android.view.ViewGroup.layout(ViewGroup.java:4362)
E/AndroidRuntime(  994): 	at android.widget.FrameLayout.onLayout(FrameLayout.java:448)
E/AndroidRuntime(  994): 	at android.view.View.layout(View.java:13754)
E/AndroidRuntime(  994): 	at android.view.ViewGroup.layout(ViewGroup.java:4362)
E/AndroidRuntime(  994): 	at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:1866)
E/AndroidRuntime(  994): 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1687)
E/AndroidRuntime(  994): 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:998)
E/AndroidRuntime(  994): 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:4212)
E/AndroidRuntime(  994): 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:725)
E/AndroidRuntime(  994): 	at android.view.Choreographer.doCallbacks(Choreographer.java:555)
E/AndroidRuntime(  994): 	at android.view.Choreographer.doFrame(Choreographer.java:525)
E/AndroidRuntime(  994): 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:711)
E/AndroidRuntime(  994): 	at android.os.Handler.handleCallback(Handler.java:615)
E/AndroidRuntime(  994): 	at android.os.Handler.dispatchMessage(Handler.java:92)
E/AndroidRuntime(  994): 	at android.os.Looper.loop(Looper.java:137)
E/AndroidRuntime(  994): 	at android.app.ActivityThread.main(ActivityThread.java:4745)
E/AndroidRuntime(  994): 	at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime(  994): 	at java.lang.reflect.Method.invoke(Method.java:511)
E/AndroidRuntime(  994): 	at com.android.intern
W/ActivityManager(  398):   Force finishing activity org.wordpress.android/.ui.posts.PostsListActivity
```

**Comment by nbradbury** [comment]
Closed (fixed) via #3105
